### PR TITLE
Fritz hoing Github actions improvements

### PR DESCRIPTION
**Pull request by ichim-david**
----
📚 Documentation preview 📚: https://volto--6120.org.readthedocs.build/

Checked out pull request https://github.com/plone/volto/pull/6108/files and fixed the issues
regarding the action run error codes for example:
Error: /home/runner/work/volto/volto/./.github/actions/pre_test_setup/action.yml (Line: 4, Col: 3): There's not enough info to determine what you meant. Add one of these properties: args, entrypoint, env, image, main, post, post-entrypoint, pre, pre-entrypoint, steps

### CODE REVIEW

**Comment by ichim-david** [comment]
Closing in favor of having commit suggestions to the original pull request https://github.com/plone/volto/pull/6108#pullrequestreview-2134173289

**Comment by ichim-david** [comment]
@stevepiercy Have another look here, this pull request still contains the work of @FritzHoing with the right attribution plus my changes since it's very cumbersome todo suggest commits especially where it's a trial an error.
Every shell: bash is needed, you can look at the failures I got after removing them https://github.com/plone/volto/pull/6120/commits/f04c7419d0e7b85f907b9c1ef7ccdebe3a662da9
and all of the commits I did until the build was clean

**Review by stevepiercy — APPROVED** [review]
Thanks for the thorough verification of extracting all the repetitive steps, including bash requirement. I especially like that you added a name to an unnamed step. It's documentation! 😁 

@plone/volto-team this is good work that you could use as an example and apply to your own CI to avoid copy-pasta mistakes. Please take a moment to review.

**Comment by sneridagh** [comment]
@ichim-david @FritzHoing Awesome work! Thanks for working on this. Could we find a better naming for the action? `pre_test_setup` is too vague for me, in anticipation that we could have more in the future. How about `node_pnpm_setup` or `setup_node_pnpm`? Ahhhh, naming, naming

**Comment by stevepiercy** [comment]
I find naming a thing by what it does, not its tool name, is usually more descriptive and long-lived. If you ever change the tool (which _never_ happens in JavaScript), you don't have to change the name.

It creates a universal test setup, so `universal_test_setup` works for me.

**Comment by ichim-david** [comment]
@sneridagh @stevepiercy I agree on the naming needs tweaking, it's for that reason that I replaced Pretest-Setup with 
name: Set up testing infrastructure when using it. Should that naming also be changed so something else?

I don't mind the universal_test_setup naming.
acceptance_test_setup?

Maybe you can come to an agreement on the naming and proposing it then, maybe talk with Eric and David and see what you guys thing makes most sense.

**Comment by sneridagh** [comment]
@stevepiercy @ichim-david the fact is that it can be used outside acceptance, in unit and code quality too. It's setting up the node version and pnpm in the right way. Also, is not universal, as we can configure other things too, like Python...

**Comment by sneridagh** [comment]
@stevepiercy and I, we discussed it and agreed on `node_env_setup`, @ichim-david opinion?

**Comment by ichim-david** [comment]
@sneridagh I'm fine with it what about the step name?

**Comment by sneridagh** [comment]
@ichim-david `Set up Node environment` ? @stevepiercy ?

**Comment by stevepiercy** [comment]
Yup. We have a winner!

**Comment by ichim-david** [comment]
@sneridagh done, let's see the testing and then a merge of main and I think it's good to go.

File: .github/actions/node_env_setup/action.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Set up testing infrastructure 
+name: Set up NodeJS environment
```
Comment (stevepiercy) [inline .github/actions/node_env_setup/action.yml:1]: This is the brand spelling. Please correct throughout.

```suggestion
name: Set up Node.js environment
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
It's Node.js per https://nodejs.org/en

**Review by stevepiercy — APPROVED** [review]

**Pull request merged**